### PR TITLE
Bump Dockerfile to go 1.9.1

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Make sure to bump Dockerfile to 1.9.1

## Motivation and Context
Additionally add Dockerfile.dev to build local development images.

## How Has This Been Tested?
Manually.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by nitisht — APPROVED** [review]
LGTM

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM alpine:3.6
+FROM golang:1.9.1-alpine3.6
```
Comment (vadmeste) [inline Dockerfile:1]: I noticed that this script install go 1.8.3 (`(16/18) Installing go (1.8.3-r0)`)

In addition, isn't changing from 'alpine:3.6'  to 'golang:1.9.1-alpine3.6' we don't have to install go anymore ?

Comment (harshavardhana) [reply Dockerfile:1]: we don't @vadmeste . let me fix that.

**Comment by vadmeste** [comment]
@harshavardhana, shouldn't we also update Docker.release ?

**Comment by harshavardhana** [comment]
> @harshavardhana, shouldn't we also update Docker.release ?

why? @vadmeste Dockerfile.release doesn't need any compiler, it is a binary only build.

**Comment by vadmeste** [comment]
> why? @vadmeste Dockerfile.release doesn't need any compiler, it is a binary only build.

Right.

**Review by vadmeste — APPROVED** [review]
LGTM

**Pull request merged**